sdcm/utils/aws_region.py

```diff
@@ -99,51 +86,101 @@ def get_availability_zones_for_instance_type(self, instance_type):
+                                           VpcId=self.sct_vpc.vpc_id, AvailabilityZone=region_az)
+        subnet_id = result["Subnet"]["SubnetId"]
+        subnet = self.resource.Subnet(subnet_id)  # pylint: disable=no-member
+        subnet.create_tags(Tags=[{"Key": "Name", "Value": subnet_name}])
+        LOGGER.info("Configuring to automatically assign public IPv4 and IPv6 addresses...")
+        self.client.modify_subnet_attribute(
+            MapPublicIpOnLaunch={"Value": True},
+            SubnetId=subnet_id
+        )
+        # for some reason boto3 throws error when both AssignIpv6AddressOnCreation and MapPublicIpOnLaunch are used
+        self.client.modify_subnet_attribute(
+            AssignIpv6AddressOnCreation={"Value": True},
+            SubnetId=subnet_id
+        )
+        LOGGER.info("'%s' with id '%s' created.", subnet_name, subnet_id)
+
+    def associate_subnet_with_route_table(self, region_az: str, index: str):
+        subnet = self.sct_subnet(region_az, subnet_index=str(index))
+        sct_route_table = self.sct_route_table(index=index)
+        response = self.client.describe_route_tables(
+            Filters=[{'Name': 'association.subnet-id', 'Values': [subnet.subnet_id]}])
+
+        # If subnet is not associated with route table or associated with main route table
+        if not (associated_route_table := response['RouteTables']):
+            self.client.associate_route_table(RouteTableId=sct_route_table.route_table_id, SubnetId=subnet.subnet_id)
+            LOGGER.info("Subnet '%s' has been associated with '%s' route table.",
+                        subnet.subnet_id, sct_route_table.route_table_id)
+            return
+
+        associated_route_table_id = associated_route_table[0].get('RouteTableId')
+        # If subnet is associated with route table but not that was created for this subnet
+        if associated_route_table_id != sct_route_table.route_table_id:
+            self.client.replace_route_table_association(
+                AssociationId=associated_route_table_id, RouteTableId=sct_route_table.route_table_id)
+            LOGGER.info("Subnet '%s' has been associated with '%s' route table.",
+                        subnet.subnet_id, sct_route_table.route_table_id)
+        # If subnet is associated with appropriate route table
+        else:
+            LOGGER.info("Subnet '%s' is already associated with '%s' route table.",
+                        subnet.subnet_id, sct_route_table.route_table_id)
 
     def create_sct_subnets(self):
-        num_subnets = len(self.availability_zones)
-        ipv4_cidrs = list(self.vpc_ipv4_cidr.subnets(6))[:num_subnets]
-        ipv6_cidrs = list(self.vpc_ipv6_cidr.subnets(8))[:num_subnets]
-        for i, az_name in enumerate(self.availability_zones):
-            self.create_sct_subnet(region_az=az_name, ipv4_cidr=ipv4_cidrs[i], ipv6_cidr=ipv6_cidrs[i])
+        begin_cidrs = 0
```

Comment:
iterators, and using next, instead of indexes, can make it a bit clearer to understand:
see the following example:
```python
from ipaddress import ip_network

vpc_ipv4_cidr = ip_network('10.12.0.0/16')
vpc_ipv6_cidr = ip_network('2001:db8::/63')

ipv4_iter = vpc_ipv4_cidr.subnets(6)
ipv6_iter = vpc_ipv6_cidr.subnets(8)

subnets_per_az = 2

for az_name in ['A', 'B', 'C']:
    for range_index in range(subnets_per_az):
        print(dict(region_az=az_name, ipv4_cidr=next(ipv4_iter), ipv6_cidr=next(ipv6_iter), subnet_index=str(range_index)))
```

Reply:
changed